File: JoystickTestboard2021/src/main/cpp/Robot.cpp
#include "Robot.h"
#include <iostream>
#include <units/math.h>

void Robot::RobotInit() {
    encoder.SetVelocityConversionFactor(wpi::math::pi * 2.0 / 60.0);
}

void Robot::RobotPeriodic() {
    std::cout << "Motor Velocity: " << encoder.GetVelocity() << std::endl;
    std::cout << "Motor Temperature: " << motor.GetMotorTemperature() << std::endl;
}

void Robot::AutonomousInit() {}

void Robot::AutonomousPeriodic() {}

void Robot::TeleopInit() {}

void Robot::TeleopPeriodic() {

   if (joystick.GetRawButtonPressed(3)) {
       motor.Set(0);
   } else if (joystick.GetRawButtonPressed(4)) {
       motor.Set(0.2);
   } else if (joystick.GetRawButtonPressed(5)) {
       motor.Set(0.3);
   } else if (joystick.GetRawButtonPressed(6)) {
       motor.Set(0.4);
   } else if (joystick.GetRawButtonPressed(7)) {
       motor.Set(0.5);
   } else if (joystick.GetRawButtonPressed(8)) {
       motor.Set(0.6);
   } else if (joystick.GetRawButtonPressed(9)) {
       motor.Set(0.7);
   } else if (joystick.GetRawButtonPressed(10)) {
       motor.Set(0.8);
   } else if (joystick.GetRawButtonPressed(11)) {
       motor.Set(0.9);
   } else if (joystick.GetRawButtonPressed(12)) {
       motor.Set(1.0);
   }
}

void Robot::DisabledInit() {}

void Robot::DisabledPeriodic() {}

void Robot::TestInit() {}

void Robot::TestPeriodic() {}

#ifndef RUNNING_FRC_TESTS
int main() {
  return frc::StartRobot<Robot>();
}
#endif

File: Robot-2022-Java-Rewrite-main/src/main/java/frc3512/robot2022/Robot.java
package frc3512.robot2022;

import edu.wpi.first.wpilibj.DriverStation;
import edu.wpi.first.wpilibj.TimesliceRobot;
import edu.wpi.first.wpilibj.livewindow.LiveWindow;
import edu.wpi.first.wpilibj2.command.Command;
import edu.wpi.first.wpilibj2.command.CommandScheduler;

public class Robot extends TimesliceRobot {
  private Command m_autonomousCommand;

  private RobotContainer m_robotContainer;

  public Robot() {
    super(0.005, 0.01);

    DriverStation.silenceJoystickConnectionWarning(true);
    LiveWindow.disableAllTelemetry();

    // Runs for 2 ms after first controller function
    schedule(() -> m_robotContainer.m_frontFlywheel.controllerPeriodic(), 0.0007);

    // Runs for 2 ms after robot periodic functions
    schedule(() -> m_robotContainer.m_backFlywheel.controllerPeriodic(), 0.0007);
  }

  @Override
  public void robotInit() {
    m_robotContainer = new RobotContainer();
  }

  @Override
  public void robotPeriodic() {
    CommandScheduler.getInstance().run();
  }

  @Override
  public void disabledInit() {}

  @Override
  public void disabledPeriodic() {}

  @Override
  public void autonomousInit() {
    m_autonomousCommand = m_robotContainer.getAutonomousCommand();

    if (m_autonomousCommand != null) {
      m_autonomousCommand.schedule();
    }
  }

  @Override
  public void autonomousPeriodic() {}

  @Override
  public void teleopInit() {
    if (m_autonomousCommand != null) {
      m_autonomousCommand.cancel();
    }
  }

  @Override
  public void teleopPeriodic() {}

  @Override
  public void testInit() {
    CommandScheduler.getInstance().cancelAll();
  }

  @Override
  public void testPeriodic() {}

  @Override
  public void simulationInit() {}

  @Override
  public void simulationPeriodic() {}
}
